docs/custom-flows/oauth-connections.mdx

```diff
@@ -5,173 +5,172 @@ description: Learn how to leverage the Clerk SDK to build completely custom OAut
+
+To allowlist a redirect URL via the Clerk Dashboard:
+1. Navigate to the [Clerk Dashboard](https://dashboard.clerk.com/last-active?path=user-authentication/social-connections).
+2. In the navigation sidebar, select **User & Authentication > Social Connections**.
+3. Scroll to the **Allowlist for mobile OAuth redirect** section and add your redirect URLs.
+
+## OAuth account transfer flows
+
+In some cases, a user may be trying to sign in with an OAuth account, but they don't have an account in your application yet. Or a user may be trying to sign up with an OAuth account, but they already have an account in your application. In these cases, Clerk provides "account transfers" which allow you to forward the user's information to the appropriate flow.
+
+The following example shows how to handle these cases in your sign-in flow. The same logic can be applied to the sign-up flow; change the `signIn.AuthenticateWithRedirect()` to `signUp.authenticateWithRedirect()`.
+
+<Tabs items={["Next.js"]}>
+  <Tab>
+  ```jsx filename="app/sign-in/[[...sign-in]]/page.tsx"
+  'use client';
+
+  import * as React from 'react';
+  import { OAuthStrategy } from '@clerk/types';
+  import { useSignIn, useSignUp } from '@clerk/nextjs';
+
+  export default function OauthSignIn() {
+    const { signIn } = useSignIn();
+    const { signUp, setActive } = useSignUp();
+
+    if (!signIn || !signUp) return null;
+
+    async function handleSignIn(strategy: OAuthStrategy) {
+      if (!signIn || !signUp) return null;
+
+      // If the user has an account in your application, but does not yet
+      // have an OAuth account connected to it, you can transfer the OAuth
+      // account to the existing user account.
+      const userExistsButNeedsToSignIn =
+        signUp.verifications.externalAccount.status === 'transferable' &&
+        signUp.verifications.externalAccount.error?.code ===
+          'external_account_exists';
+
+      if (userExistsButNeedsToSignIn) {
+        const res = await signIn.create({ transfer: true });
+
+        if (res.status === 'complete') {
+          setActive({
+            session: res.createdSessionId,
+          });
+        }
+      }
+
+      // If the user has an OAuth account but does not yet
+      // have an account in your app, you can create an account
+      // for them using the OAuth information.
+      const userNeedsToBeCreated =
+        signIn.firstFactorVerification.status === 'transferable';
+
+      if (userNeedsToBeCreated) {
+        const res = await signUp.create({
+          transfer: true,
+        });
+
+        if (res.status === 'complete') {
+          setActive({
+            session: res.createdSessionId,
+          });
+        }
+      } else {
+        // If the user has an account in your application
+        // and has an OAuth account connected to it, you can sign them in.
+        signInWith(strategy);
+      }
+    }
+
+    const signInWith = (strategy: OAuthStrategy) => {
```

Comment:
nit: Would love to see this moved up so the function is defined before its called.